hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/process/ConsumerProcessSupervisor.java

```diff
@@ -129,7 +130,8 @@ private void processSignal(Signal signal) {
                 break;
             case KILL_UNHEALTHY:
                 onConsumerProcess(signal, consumerProcess -> {
-                    taskQueue.offer(signal.createChild(START, clock.millis(), consumerProcess.getConsumer()));
+                    Consumer consumerCopy = consumerProcess.getConsumer().createCopy();
```

Comment:
Here the idea is to create new immutable object of Consumer so we avoid situation when there are two alive ConsumerProcesses using the same Consumer object